src/pages/workspaces/workspace/analysis/cost-utils.ts

```diff
@@ -19,56 +36,71 @@ const dataprocCost = (machineType, numInstances) => {
 }
 
 const getHourlyCostForMachineType = (machineTypeName, region, isPreemptible) => {
-  const { cpu, memory } = _.find({ name: machineTypeName }, machineTypes) || {}
-  const { n1HourlyCpuPrice, preemptibleN1HourlyCpuPrice, n1HourlyGBRamPrice, preemptibleN1HourlyGBRamPrice } = _.find({ name: _.toUpper(region) },
+  const { cpu, memory } = _.find({ name: machineTypeName }, machineTypes) || { cpu: 0, memory: 0 }
+  const { n1HourlyCpuPrice = 0, preemptibleN1HourlyCpuPrice = 0, n1HourlyGBRamPrice = 0, preemptibleN1HourlyGBRamPrice = 0 } = _.find({ name: _.toUpper(region) },
     regionToPrices) || {}
   return isPreemptible ?
     (cpu * preemptibleN1HourlyCpuPrice) + (memory * preemptibleN1HourlyGBRamPrice) :
     (cpu * n1HourlyCpuPrice) + (memory * n1HourlyGBRamPrice)
 }
 
 const getGpuCost = (gpuType, numGpus, region) => {
-  const prices = _.find({ name: region }, regionToPrices)
+  const prices = _.find({ name: region }, regionToPrices) || {}
   // From a type like 'nvidia-tesla-t4', look up 't4HourlyPrice' in prices
   const price = prices[`${_.last(_.split('-', gpuType))}HourlyPrice`]
   return price * numGpus
 }
 
-export const runtimeConfigBaseCost = config => {
-  const {
-    cloudService, masterMachineType, masterDiskSize, numberOfWorkers, workerMachineType, workerDiskSize, bootDiskSize, computeRegion
-  } = normalizeRuntimeConfig(config)
+const getDefaultIfUndefined = (input: number | undefined, alternative: number): number => input ? input : alternative
 
-  const isDataproc = cloudService === cloudServices.DATAPROC
+// This function deals with runtimes that are paused
+// All disks referenced in this function are boot disks (aka the disk google needs to provision by default for OS storage)
+// The user pd cost for a runtime is calculated elsewhere
+export const runtimeConfigBaseCost = (config: GoogleRuntimeConfig): number => {
```

Comment:
This function was an intolerable abomination and a great illustration of Typescript's strengths.

It essentially ignored the fact that the backend model had 3 possible implementation with different fields, and tried to transform any of the 3 into a common model with the same fields (`normalizeRuntimeConfig`, now removed). There were several bugs with this uncovered when I typed RuntimeConfigs.